src/named_tuple.cr

```diff
@@ -38,13 +38,16 @@ struct NamedTuple
   end
 
   # Creates a named tuple from the given hash, with elements casted to the given types.
+  # Here the Int32 | String union is cast to Int32.
   #
   # ```
-  # NamedTuple(foo: String, bar: Int64).from({:foo => "world", :bar => 2})       # => {foo: "world", bar: 2}
-  # NamedTuple(foo: String, bar: Int64).from({"foo" => "world", "bar" => 2})     # => {foo: "world", bar: 2}
-  # NamedTuple(foo: String, bar: Int64).from({:foo => "world", :bar => 2}).class # => {foo: String, bar: Int64}
-  # ```
+  # num_or_str : Int32 | String
+  # num_or_str = 42
+  # NamedTuple(name: String, val: Int32).from({"name" => "number", "val" => num_or_str}) # => {name: "number", val: 42}
   #
+  # num_or_str = "a string"
+  # NamedTuple(name: String, val: Int32).from({"name" => "number", "val" => num_or_str}) # =>  can't cast String to Int32
```

Comment:
There is a double space after `=>`. 

To keep consistency, I think the text should be: 
```
# raises TypeCastError (cast from String to Int32 failed)
```

check PR #3820 and https://github.com/maiha/comment-spec.cr